sdk/CMakeLists.txt

```diff
@@ -180,12 +161,17 @@ add_custom_command(
     ${_program_schema__include_dir}/executorch/sdk/etdump
     ${_etdump_schema__srcs}
   COMMAND rm -f ${_etdump_schema_cleanup_paths}
-  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/sdk
   DEPENDS ${_etdump_schema_gen_dep}
   COMMENT "Generating etdump headers"
-  VERBATIM
 )
 
+add_custom_target(
+  etdump_schema_generated
+  DEPENDS ${_etdump_schema__outputs}
+)
+
+add_dependencies(etdump_schema etdump_schema_generated)
```

Comment:
Could these DEPENDS be directly on `etdump_schema` instead of creating this extra target?

Up above we already say `add_library(etdump_schema INTERFACE ${_etdump_schema__outputs})` -- is that enough to add the dependencies? Or maybe you could also add a DEPENDS section to that same `add_library`?

If this custom_target is necessary, please arrange the code so that it's closer to the `add_library(etdump_schema, ...)`, and add a comment explaining why these deps can't be added directly to the library target.

Reply:
Yes, it can. Thanks for the hint, that's much cleaner.